fashion-hyperparameter/optimize_class.py

```diff
@@ -0,0 +1,204 @@
+
+
+class FlowRunner:
+    def __init__(self, index_document_generator, query_document_generator,
+                       index_batch_size, query_batch_size,
+                       env_yaml = None,
+                       workspace_env = 'JINA_WORKSPACE',
+                       overwrite_workspace=False):
+
+        self.index_document_generator = index_document_generator
+        self.query_document_generator = query_document_generator
+        self.index_batch_size = index_batch_size
+        self.query_batch_size = query_batch_size
+        self.env_yaml = env_yaml
+        self.workspace_env = workspace_env
+        self.overwrite_workspace = overwrite_workspace
+
+    @staticmethod
+    def clean_workdir(workspace):
+        if workspace.exists():
+            shutil.rmtree(workspace)
+            print(colored('--------------------------------------------------------', 'red'))
+            print(colored('-------------- Existing workspace deleted --------------', 'red'))
+            print(colored('--------------------------------------------------------', 'red'))
+            print(colored('WORKSPACE: ' + str(workspace), 'red'))
+
+    def _load_env(self):
+        if self.env_yaml:
+            yaml = YAML(typ='safe')
+            self.env_parameters = yaml.load(open(self.env_yaml))
+            for environment_variable, value in self.env_parameters.items():
+                os.environ[environment_variable] = str(value)
+            logger.info('Environment variables loaded')
+        else:
+            logger.info('Cannot load environment variables as no env_yaml passed')
+
+    def run_indexing(self, index_yaml, workspace=None):
+        self._load_env()
+        workspace = Path(os.environ.get(self.workspace_env, workspace))
+
+        if workspace.exists():
+            if self.overwrite_workspace:
+                FlowRunner.clean_workdir(workspace)
+            else:
+                print(colored('--------------------------------------------------------', 'cyan'))
+                print(colored('----- Workspace already exists. Skipping indexing. -----', 'cyan'))
+                print(colored('--------------------------------------------------------', 'cyan'))
+                return
+
+        self.index_document_generator, index_document_generator = tee(self.index_document_generator)
+
+        with Flow().load_config(index_yaml) as f:
+            f.index(index_document_generator, batch_size=self.index_batch_size)
+
+    def run_querying(self, query_yaml, callback):
+        self._load_env()
+        self.query_document_generator, query_document_generator = tee(self.query_document_generator)
+
+        with Flow().load_config(query_yaml) as f:
+            f.search(
+                query_document_generator,
+                batch_size=self.query_batch_size,
+                output_fn=callback
+            )
+
+    def run_evaluation(self, index_yaml, query_yaml, workspace, evaluation_callback):
```

Comment:
What if I want to evaluate an indexing flow? Remember that evaluation is agnostic to the flow u run. One can evaluate anything